src/pow.rs

```diff
@@ -0,0 +1,1314 @@
+            #[doc = concat!("let a = ", stringify!($ty), "::from(0xdeadbeef);")]
+            #[doc = concat!("let b = ", stringify!($ty), "::from_le([0xdeadbeef, 0, 0, 0]);")]
+            /// assert_eq!(a, b)
+            /// ```
+            #[doc = concat!("", stringify!($ty), ";")]
+            #[inline]
+            pub fn from_le(array: [u64; 4]) -> $ty { $ty(Uint256::from_le((array))) }
+
+            #[doc = concat!("Creates a [`", stringify!($ty), "`] from a big-endian byte array.")]
+            #[inline]
+            pub fn from_be_bytes(bytes: [u8; 32]) -> $ty { $ty(Uint256::from_be_bytes((bytes))) }
+
+            #[doc = concat!("Creates a [`", stringify!($ty), "`] from a little-endian byte array.")]
+            #[inline]
+            pub fn from_le_bytes(bytes: [u8; 32]) -> $ty { $ty(Uint256::from_le_bytes((bytes))) }
+
+            #[doc = concat!("Creates a [`", stringify!($ty), "`] from a big-endian byte slice.")]
+            #[inline]
+            pub fn from_be_slice(bytes: &[u8]) -> Result<$ty, ParseLengthError> {
+                let u = Uint256::from_be_slice(bytes)?;
+                Ok($ty(u))
+            }
+
+            #[doc = concat!("Creates a [`", stringify!($ty), "`] from a little-endian byte slice.")]
+            #[inline]
+            pub fn from_le_slice(bytes: &[u8]) -> Result<$ty, ParseLengthError> {
+                let u = Uint256::from_le_slice(bytes)?;
+                Ok($ty(u))
+            }
+
+            /// Converts `self` to a big-endian array.
+            ///
+            /// # Examples
+            ///
+            /// ```
+            #[doc = concat!("use bitcoin::", stringify!($ty), ";")]
+            #[doc = concat!("let a = ", stringify!($ty), "::from(0xdeadbeef);")]
+            /// assert_eq!(a.to_be(), [0, 0, 0, 0xdeadbeef]);
+            /// ```
+            #[inline]
+            pub fn to_be(self) -> [u64; 4] { self.0.to_be() }
+
+            /// Converts `self` to a little-endian array.
+            ///
+            /// # Examples
+            ///
+            /// ```
+            #[doc = concat!("use bitcoin::", stringify!($ty), ";")]
+            #[doc = concat!("let a = ", stringify!($ty), "::from(0xdeadbeef);")]
+            /// assert_eq!(a.to_le(), [0xdeadbeef, 0, 0, 0]);
+            /// ```
+            #[inline]
+            pub fn to_le(self) -> [u64; 4] { self.0.to_le() }
+
+            /// Converts `self` to a big-endian byte array.
+            #[inline]
+            pub fn to_be_bytes(self) -> [u8; 32] { self.0.to_be_bytes() }
+
+            /// Converts `self` to a little-endian byte array.
+            #[inline]
+            pub fn to_le_bytes(self) -> [u8; 32] { self.0.to_le_bytes() }
+        }
+
+        impl From<u64> for $ty {
+            #[inline]
+            fn from(x: u64) -> Self { $ty(Uint256::from(x)) }
+        }
+
+        impl fmt::Display for $ty {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::Display::fmt(&self.0, f) }
```

Comment:
The `Debug`/`Display` impls are a bit messy. One would expect that `Display` of `Uint256` display in decimal which it doesn't do. It'd be clearer to implement `LowerHex` instead (with only `0x` added if `f.alternate()`) and delegate to that here instead. Also would be nice to handle alignment and such.

Reply:
Aight, final task after lunch!  Thanks for the review @Kixunil, very thorough and educational, I appreciate your efforts.